.github/workflows/new_release.yml

```diff
@@ -1,36 +1,54 @@
-# This is a basic workflow to help you get started with Actions
-
 name: CI new release
 
-# Controls when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the "master" branch
-  push:
-    branches: [ "master" ]
-  pull_request:
-    branches: [ "master" ]
-
-  # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
+    inputs:
+      version:
+        description: Released new version of Realm (0.X.0+beta).
+        required: true
+        type: string
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
-    # The type of runner that the job will run on
+   prepare-pr:
     runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v3
 
-      # Runs a single command using the runners shell
-      - name: Run a one-line script
-        run: echo Hello, world!
+      - name: Update realm_dart in pubspec.yaml
+        uses: jacobtomlinson/gha-find-replace@v2
+        with:
+          find: 'realm_dart: .*'
+          replace: 'realm_dart: ${{ inputs.version }}'
+          include: '**pubspec.yaml'
+
+      - name: Update realm in pubspec.yaml
+        uses: jacobtomlinson/gha-find-replace@v2
+        with:
+          find: 'realm: .*'
+          replace: 'realm: ${{ inputs.version }}'
+          include: '**pubspec.yaml'
 
-      # Runs a set of commands using the runners shell
-      - name: Run a multi-line script
+      - name: Setup Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          channel: 'stable'
+
+      - name: Install dependencies, run generator, run realm on dart
+        working-directory: .
         run: |
-          echo Add other actions to build,
-          echo test, and deploy your project.
+          chmod +x .github/workflows/ci.sh
```

Comment:
This is showing up as a change in PRs - should we just change the permissions of the file as part of this PR?

Reply:
I don't understand what you mean.

Reply:
If you look at the generated PR, you can see that `ci.sh` changed: https://github.com/realm/realm-dart-samples/pull/23/files#diff-d6c9574a668c2cd83872bb7b958b5782eb26ffe482dc77a7dd03bf6ccf2e6ee2. It went from 644 to 755 due to the `chmod +x` we're doing here. My suggestion was to change the permissions of `ci.sh` here so that it's 755 already, which would mean we can remove this line.

Reply:
I merged the new mode file to ds/ci_after_release_workflow

Reply:
Cool, then the chmod +x should not be needed anymore.